### ISSUE TITLE
CharacterBody3D get stuck in StaticBody3D with BoxShape3Ds when moving with move_and_slide

### ISSUE THREAD

**Issue by alfredbaudisch** [godotengine/godot#75647]
### Godot version

4.0.2

### System information

Windows 10

### Issue description

CharacterBody3D moving with `move_and_slide` gets stuck between a StaticBody3D with 2x BoxShape3D and keep "bouncing/shaking" excessively when trying to move:

https://user-images.githubusercontent.com/248383/229829759-382d09d3-5047-48ed-8e9e-587fc88df493.mp4

This is different from https://github.com/godotengine/godot/issues/75399 and https://github.com/godotengine/godot/issues/70453 as those ones are related to Convex Hulls and ConcavePolygonShape3D.

Collision setup:
![imagen](https://user-images.githubusercontent.com/248383/229829223-046d9afc-3dd7-4bbf-a120-5b95a40f96c1.png)

![imagen](https://user-images.githubusercontent.com/248383/229829902-efcb25be-11f6-4666-87ea-3b70f6f4a873.png)

![imagen](https://user-images.githubusercontent.com/248383/229830092-6cee5300-f25d-4ebd-80ae-567f8eb252fe.png)

### Steps to reproduce

- Add a StaticBody3D and 2 BoxShape3D close to one another
- Move with a CharacterBody3D `move_and_slide` against the StaticBody3D
- Character will get stuck and bouncing in place

### Minimal reproduction project

Run project and move forward against the ramp:
[StuckGeometryCollisionBoxShape3D.zip](https://github.com/godotengine/godot/files/11149871/StuckGeometryCollisionBoxShape3D.zip)

**Comment by viksl** [comment]
I think this is basically this: https://github.com/godotengine/godot/issues/75537, it's also the same issues as i have which alfred linked above,

**Comment by alfredbaudisch** [comment]
@viksl my report is when colliding with StaticBody3D the one you linked with RigidBody3D. So both are worth keeping and validating.

**Comment by viksl** [comment]
@alfredbaudisch Yeah, but it happens with any body, static, rigid, character all of them display the same issue it seems so far.

**Comment by comradecid** [comment]
this also occurs when interacting two 2d sprites using move_and_slide

**Comment by StarboyCZ** [comment]
Hi, I started with official tutorial (make first 3d game) and yes, I stuck with my character with CapsuleShape collisionbox3D with static body with CylinderShape collisionbox3D, in certain part of cylinder with certain scale of cylinder and capsule. Then if I am stuck, then I cannot go even directly
 from that object or sometimes I stuck completely in any direction.
 Thank you for solving this.
 
![Snímek obrazovky 2024-02-06 090719](https://github.com/godotengine/godot/assets/36074620/88a715d5-1612-4cd9-b6e7-d9e9273cd477)

Stuck:
![Untitled-1](https://github.com/godotengine/godot/assets/36074620/3d906b20-db5d-4a05-959d-edf7ac336629)
![Untitled-2](https://github.com/godotengine/godot/assets/36074620/72b892a0-f8bc-4760-9676-3d10b64af85c)

**Comment by StarboyCZ** [comment]
I think I solve it: I change the value of Safe Margin from default 0.001 (lowest possible) to 0.002 m and it work now. So it has to be a bug.
![Snímek obrazovky 2024-02-06 145401](https://github.com/godotengine/godot/assets/36074620/eca3c532-91ee-4e46-9d8e-390104de8d6f)
